curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/68140ab94fd0ea19c62ca87e.md

```diff
@@ -0,0 +1,268 @@
+---
+id: 68140ab94fd0ea19c62ca87e
+title: Step 21
+challengeType: 0
+dashedName: step-21
+---
+
+# --description--
+
+At the end of the `useEffect`, return a cleanup function that clears the timeout using `clearTimeout(timeoutId)`. This prevents multiple delayed fetches from stacking up.
+
+# --hints--
+
+You should return `clearTimeout(timeoutId)` at the end of the `useEffect`.
```

Comment:
the current wording makes it seem like I need to return this 

```js
return clearTimeout(timeoutId)
```

it should be clear that you are looking for an arrow function that returns that